### ISSUE TITLE
conan export fails in Windows Docker container

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#2439]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

as for conan 1.0.4, the following command fails:
`conan export . username/channel`
with the following output:
```
Traceback (most recent call last):
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\command.py", line 1099, in run
    method(args[0][1:])
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\command.py", line 635, in export
    keep_source=args.keep_source)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\conan_api.py", line 63, in wrapper
    return f(*args, **kwargs)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\conan_api.py", line 537, in export
    self._manager.export(conanfile_path, name, version, user, channel, keep_source)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\manager.py", line 156, in export
    self._user_io.out, self._search_manager, self._client_cache)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\cmd\export.py", line 43, in cmd_export
    _export_conanfile(conanfile_path, output, client_cache, conanfile, conan_ref, keep_source)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\cmd\export.py", line 88, in _export_conanfile
    output)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\cmd\export.py", line 172, in _execute_export
    copier(pattern, links=True, excludes=excluded_sources)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\file_copier.py", line 78, in __call__
    self._link_folders(src, dst, link_folders)
  File "C:\Program Files (x86)\Python36-32\lib\site-packages\conans\client\file_copier.py", line 138, in _link_folders
    if os.path.exists(os.path.join(dst, dest_dir)):
  File "C:\Program Files (x86)\Python36-32\lib\genericpath.py", line 19, in exists
    os.stat(path)
ValueError: stat: embedded null character in path

ERROR: stat: embedded null character in path
['conan', 'export', '.', 'company/ci']
Traceback (most recent call last):
  File "C:\host\scripts\conan-build.py", line 178, in <module>
    run(['conan', 'export', '.', channel], args)
  File "C:\host\scripts\conan-build.py", line 39, in run
    subprocess.check_call(cmd, env=env)
  File "C:\Program Files (x86)\Python36-32\lib\subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['conan', 'export', '.', 'company/ci']' returned non-zero exit status 1.
```
it happens on Windows Server 2016 host which is running docker container with base image **microsoft/windowsservercore**.

further investigation shows it happens only if `conan export` was run from the docker volume, e.g. C:\Host in my case, which is also the directory containing recipe of jsoncpp (its conanfile.py).

from the ipython output:
```
In [1]: import os

In [2]: os.path.islink('C:\\host')
Out[2]: True

In [3]: os.readlink('C:\\host')
Out[3]: '43\x00\x00????\x00\x00????????????\ud860???????????????\ud860?\ud860?????\x01\x00\ud860???????????
'

In [4]:
```

powershell confirms it's true symlink:
```
PS C:\host> Get-Item C:\host | Select-Object -ExpandProperty Target
\ContainerMappedDirectories\D2B648BB-55A4-46A2-ABA0-97F7E4ED1A43
```

workaround exists - do not run conan export from the volume root, use some sub-directory instead.

related python issue: https://bugs.python.org/issue29248

**Comment by SSE4** [comment]
fixed in upstream https://github.com/python/cpython/pull/5577

**Comment by lasote** [comment]
What is the current state of this? What is released already in any python version?

**Comment by SSE4** [comment]
yes, it was released in python 3.7.0.
